packages/toolkit/connector-kit/src/types/oauth2.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * OAuth 2.0 client authentication methods.
+ */
+export enum ClientAuthenticationMethod {
+  ClientSecretBasic = 'client_secret_basic',
+  ClientSecretPost = 'client_secret_post',
+  ClientSecretJwt = 'client_secret_jwt',
+}
+
+/*
+ * Enumeration of algorithms supported for JWT signing when using client secrets.
+ *
+ * These "HS" algorithms (HMAC using SHA) are specifically chosen for scenarios where the
+ * client authentication method is 'client_secret_jwt'. HMAC algorithms utilize the
+ * client_secret as a shared symmetric key to generate a secure hash, ensuring the integrity
+ * and authenticity of the JWT.
+ *
+ * Other types of algorithms, such as RSASSA (RS256, RS384, RS512) or ECDSA (ES256, ES384, ES512),
+ * utilize asymmetric keys, are complex and requires secure key management infrastructure.
+ *
+ * In the 'client_secret_jwt' context, where simplicity and symmetric key usage are preferred for
+ * straightforward validation by the authorization server without the need to manage or distribute
+ * public keys, HMAC algorithms are more suitable.
+ */
+export enum ClientSecretJwtSigningAlgorithm {
+  HS256 = 'HS256', // HMAC using SHA-256 hash algorithm
```

Comment:
use the JSDoc format so we can see the comment in the VSCode while using the enum value.

```suggestion
  /** HMAC using SHA-256 hash algorithm. */
  HS256 = 'HS256',
```

Reply:
Updated.